### PR TITLE
Rework on VX (should fix this #4093)

### PR DESCRIPTION
**Pull request by xarkes**

### CODE REVIEW

**Comment by alvarofe** [comment]
When there is only one choice and pressing enter It would be great to go to that addr. 

Could you provide the same experience as with `Vx`, showing a snippet, moving around with hj and selecting with enter?

**Comment by xarkes** [comment]
Alright

File: libr/core/visual.c [hunk 1]
```diff
@@ -937,25 +942,24 @@ R_API int r_core_visual_xrefs_X(RCore *core) {
...
-		if (IS_DIGIT (ch)) {
-			refi = r_list_get_n (fun->refs, ch - 0x30);
-			if (refi) {
-				r_core_cmdf (core, "s 0x%"PFMT64x, refi->addr);
-				ret = 1;
-			}
-		}
+
+	if (IS_DIGIT (ch) && (ch - 0x30) <= count) {
+		r_core_cmdf (core, "s 0x%"PFMT64x, references[ch - 0x30]);
```
Comment (radare) [inline libr/core/visual.c]: there's no need to use cmdf for just a seek. what about using r_core_seek() instead?

File: libr/core/visual.c [hunk 2]
```diff
@@ -914,18 +914,23 @@ R_API int r_core_visual_xrefs_X(RCore *core) {
 		addr += core->print->cur;
 	}
 
+	r_cons_clear00 ();
+	r_cons_gotoxy (1, 1);
+	r_cons_printf ("[GOTO REF]> \n");
+
+	ut64 references[10];
```
Comment (radare) [inline libr/core/visual.c]: better define this at the begining of the function, and if you do so initialize with = {0};

**Comment by alvarofe** [comment]
Merged. Thanks though I spot an issue in Vx, that I will open and assign it to you since it should be easy for you to fix it :)

**Pull request closed without merging**